lib/elixir/lib/option_parser.ex

```diff
@@ -492,4 +556,22 @@ defmodule OptionParser do
       :unknown
     end
   end
+
+  defp format_errors(errors) do
+    details = Enum.map_join(errors, ". ", &format_error(&1))
+    total = Enum.count(errors)
+    error_pluralize = pluralize(total, "error", "errors")
+    "You have #{total} #{error_pluralize}! #{details}"
```

Comment:
Personally I'm not a fan of the colloquial style of this error message (I'm talking about the `You have 3 errors!` part). What about something like this:

```
3 errors found.

  * ...
  * ...
```

wdyt?